pkg/resource/deploy/target.go

```diff
@@ -29,6 +29,15 @@ type Target struct {
 	Snapshot     *Snapshot        // the last snapshot deployed to the target.
 }
 
+const (
+	nodejsDynamicProviderPackage = "pulumi-nodejs"
+	pythonDynamicProviderPackage = "pulumi-python"
+)
+
+func isDynamicProvider(pkg tokens.Package) bool {
+	return pkg == tokens.Package(nodejsDynamicProviderPackage) || pkg == tokens.Package(pythonDynamicProviderPackage)
+}
```

Comment:
I think it might be worth documenting these constants and this function briefly.